src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -278,7 +409,14 @@ private void OnCollectionChanged(NotifyCollectionChangedAction action, object it
         /// </summary>
         private void OnCollectionChanged(NotifyCollectionChangedAction action, object oldItem, object newItem, int index)
         {
-            OnCollectionChanged(new NotifyCollectionChangedEventArgs(action, newItem, oldItem, index));
+            if (oldItem is IList && newItem is IList)
```

Comment:
Same comment as above.

Reply:
Removed unnecessary if statement after we added additional overloads as recommended.